### PR TITLE
Move extensive task docstring to getting-started.md

### PR DESCRIPTION
**Pull request by elliotgunton**
* Also fix "Examples:" sections breaking mkdocs rendering

**Pull Request Checklist**
- [x] Fixes bad doc rendering
- [ ] ~Tests added~
- [x] Documentation/examples added
- [x] [Good commit messages](https://cbea.ms/git-commit/) and/or PR title

**Description of PR**
Currently, mkdocs is having trouble rendering the doc website

This PR fixes some of the docstrings escaping the rendering which results in sections like 
image
image

### CODE REVIEW

File: src/hera/workflows/task.py [hunk 1]
```diff
@@ -49,91 +49,7 @@ class Task(
...
-
-    As a diagram:
-
-    ```
-    A   B
-    | X |
-    C   D
-    ```
-    """
+    """Task is used to run a given template within a DAG. Must be instantiated under a DAG context."""
```
Comment (dejamiko) [inline src/hera/workflows/task.py:52]: Maybe it's worth linking back to the doc?

Comment (elliotgunton) [reply src/hera/workflows/task.py:52]: I would rather keep the inline docstrings self-contained as the getting started guide structure may change/evolve so we don't want to have to update lots of docstrings in that case.

**Review by flaviuvadan — APPROVED** [review]
🎖️

**Comment by elliotgunton** [comment]
image 
image


ToC now fixed all the way down 😎

Might just update links as they're not clickable which I think need to be markdown `[text](link)` style

image

**Review by samj1912 — APPROVED** [review]

**Pull request merged**